### DISCUSSION TITLE
Migrate to KeyMap?

### DISCUSSION THREAD

**Discussion by RobertFischer** [brendanhay/amazonka#816] [Ideas]
Let's not hold up the Amazonka 2 release for it, but whenever we get around to Amazonka 3, would you be willing to consider moving from `HashMap Text` to Aeson's `KeyMap`, at least for the DynamoDB requests and responses?  I have no idea what this would look like in terms of a code change, but it'd almost certainly be more efficient and easier to work with now that we're using Aeson 2.

**Comment by endgame** [comment]
I think this thread predates the custom `AttributeValue` type in `amazonka-dynamodb`, and for other services I'm not sure that `KeyMap` makes much sense (though I'm happy to have that discussion).
